public/static/js/projects.js

```diff
@@ -379,14 +386,34 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location,
       $scope.addGithubAlias();
       if($scope.github_alias) {
         if($scope.github_alias.variant) {
-          $scope.invalidPatchDefinitionMessage = "Missing task regex";
+          $scope.invalidGitHubPatchDefinitionMessage = "Missing task regex";
         }else if($scope.github_alias_task) {
-          $scope.invalidPatchDefinitionMessage = "Missing variant regex";
+          $scope.invalidGitHubPatchDefinitionMessage = "Missing variant regex";
+        }
+        return;
+      }
+    }
+
+    if ($scope.commit_queue_alias) {
```

Comment:
I think the pattern here is confusing. I would first validate that the data structure the controls are bound to, and if valid add it to the structure that is saved

Reply:
I think it should be good now.